---
title: "Hackathon prep"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Every transcript needs the following features:
N reads
length of all reads (Separate file)
length of all polya tails (Separate file)
slide parameters for length: max length, proportion at max length
slide parameters for polya: max length, proportion at max length
annotated transcript length, UTR3, UTR5, exonic length, CDS length
GC content

Datasets:
mmu 3T3 Mion, 3T3 PION
hsa HeLa

Output:
half-life

Datasets:
Sharova et al - ESC mice
Eisen et al - 3t3 mice
Tani et al - HeLa human
Supplementary from Tani et al - various human


Data cleaning To do:
1. Decide on which features to use and merge them into one table
1.5. Translate transcript data to gene data
2. Extract relevant columns from the excel sheets to make validation data
3. Merge features and validation by gene names

Data on the LGG server:
Samples:
```
["hsa_dRNA_HeLa_DRB_0h_1",
"mmu_dRNA_3T3_mion_1",
"mmu_dRNA_3T3_PION_1"]
```

bam:
`/home/leect/workspace/slide/samples/{sample}/align/reads.sanitize.noribo.toTranscriptome.sorted.bam`

polya lengths:
`/home/leect/workspace/slide/analysis/slide_polya/{sample}/polyA_read_length.tab`

read lengths:
`/home/leect/workspace/slide/analysis_old/length/{sample}/nanoplen_read_length_non_zero_ids.tab`

Cleaning Tani excel
```{r}
tani = readxl::read_xls("data/Tani/Tani_Supp_Tables_revised2.xls",
                        sheet = "Table S1", range = "A4:D10309"
                        )
colnames(tani) = c("RefSeq.mRNA.ID","length","RPKM","t5")
tani = tani[!is.na(tani$t5) & tani$t5 != 24,]

tani = lapply(1:nrow(tani), function(i) {
    x = tani[i,]
    a = strsplit(x$RefSeq.mRNA.ID,",")[[1]]
    d = data.frame(RefSeq.mRNA.ID = a)
    return(cbind(d, x[,-1]))
})
tani = do.call(rbind, tani)


ensembl = biomaRt::useMart("ensembl",dataset="hsapiens_gene_ensembl")
hsa_mart_export = biomaRt::getBM(attributes = c("refseq_mrna", "ensembl_transcript_id"), 
                                 filters = "refseq_mrna",
                                 values = tani$RepName, mart = ensembl, useCache = F)
colnames(hsa_mart_export)[2] ="Transcript.stable.ID"

colnames(hsa_mart_export)[1] = "RefSeq.mRNA.ID"
tani = merge(tani, hsa_mart_export, by = "RefSeq.mRNA.ID")
write.table(tani, "data/Tani/tani_halflives.txt", quote = F, sep = "\t", row.names = F, col.names = F)
```

Cleaning Eisen excel
```{r}
Eisen = readxl::read_xlsx("data/Eisen/1-s2.0-S1097276519308962-mmc2.xlsx")

Eisen = Eisen[-1, c(1,2,13)]

write.table(Eisen, "data/Eisen/Eisen_HL_cleaned.txt", quote = F, sep = "\t", row.names = F, col.names = F)
```